### PR TITLE
[dev] s2anet add anno convert to rbox

### PR DESCRIPTION
**Pull request by liuhuiCNN**
(1) add txt convert to rbox script
(2) add generate test result script
(3) add loss weight params
(4) dy2st

### CODE REVIEW

File: dataset/dota_coco/dota_generate_test_result.py [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+    'plane', 'baseball-diamond', 'bridge', 'ground-track-field',
+    'small-vehicle', 'large-vehicle', 'ship', 'tennis-court',
+    'basketball-court', 'storage-tank', 'soccer-ball-field', 'roundabout',
+    'harbor', 'swimming-pool', 'helicopter', 'container-crane'
+]
+
+
+def Intersection(g, p):
+    """
+    Intersection.
```
Comment (heavengate) [inline dataset/dota_coco/dota_generate_test_result.py]: 写详细一点吧，另外变量名最好改一下，函数名不用小写开头

File: dataset/dota_coco/dota_generate_test_result.py [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+
+
+def nmsbynamedict(nameboxdict, nms, thresh):
+    """
+    Args:
+        nameboxdict: nameboxdict
+        nms:   nms
+        thresh: nms threshold
+
+    Returns:
```
Comment (heavengate) [inline dataset/dota_coco/dota_generate_test_result.py]: 补充下吧

File: ppdet/modeling/bbox_utils.py [hunk 3]
```diff
@@ -299,8 +299,8 @@ def delta2rbox(Rrois,
     :param wh_ratio_clip:
     :return:
     """
-    means = paddle.to_tensor(means)
-    stds = paddle.to_tensor(stds)
+    #means = paddle.to_tensor(means)
+    #stds = paddle.to_tensor(stds)
```
Comment (heavengate) [inline ppdet/modeling/bbox_utils.py]: delete

File: ppdet/modeling/bbox_utils.py [hunk 4]
```diff
@@ -391,15 +391,15 @@ def bbox_decode(bbox_preds,
     return:
         bboxes: [N,H,W,5]
     """
-    means = paddle.to_tensor(means)
-    stds = paddle.to_tensor(stds)
+    #means = paddle.to_tensor(means, dtype='float32')
+    #stds = paddle.to_tensor(means, dtype='float32')
```
Comment (heavengate) [inline ppdet/modeling/bbox_utils.py]: delete

File: ppdet/modeling/bbox_utils.py [hunk 5]
```diff
@@ -391,15 +391,15 @@ def bbox_decode(bbox_preds,
...
     num_imgs, H, W, _ = bbox_preds.shape
     bboxes_list = []
     for img_id in range(num_imgs):
         bbox_pred = bbox_preds[img_id]
         # bbox_pred.shape=[5,H,W]
         bbox_delta = bbox_pred
-        anchors = paddle.to_tensor(anchors)
+        #anchors = paddle.to_tensor(anchors.astype(np.float32), dtype='float32')
```
Comment (heavengate) [inline ppdet/modeling/bbox_utils.py]: delete

File: ppdet/modeling/bbox_utils.py [hunk 6]
```diff
@@ -512,8 +512,15 @@ def rbox2poly(rrects):
     poly:[x0,y0,x1,y1,x2,y2,x3,y3]
     """
     polys = []
-    for rrect in rrects:
-        x_ctr, y_ctr, width, height, angle = rrect[:5]
+    rrects = rrects.numpy()
+    for i in range(rrects.shape[0]):
+        rrect = rrects[i]
+        # x_ctr, y_ctr, width, height, angle = rrect[:5]
```
Comment (heavengate) [inline ppdet/modeling/bbox_utils.py]: 用这句吧，下面那个太繁琐了

File: ppdet/modeling/heads/s2anet_head.py [hunk 7]
```diff
@@ -453,11 +455,22 @@ def forward(self, feats):
...
+                np.array(
+                    self.target_stds, dtype=np.float32), dtype='float32')
+            pd_init_anchors = paddle.to_tensor(
+                np.array(
+                    init_anchors, dtype=np.float32), dtype='float32')
             refine_anchor = bbox_utils.bbox_decode(
-                fam_reg.detach(), init_anchors, self.target_means,
-                self.target_stds)
+                fam_reg1, pd_init_anchors, pd_target_means, pd_target_stds)
+            #refine_anchor = bbox_utils.bbox_decode(
```
Comment (heavengate) [inline ppdet/modeling/heads/s2anet_head.py]: delete

File: dataset/dota_coco/dota_generate_test_result.py (hunk 1 above)
Comment (liuhuiCNN) [reply dataset/dota_coco/dota_generate_test_result.py]: done

File: ppdet/modeling/bbox_utils.py (hunk 3 above)
Comment (liuhuiCNN) [reply ppdet/modeling/bbox_utils.py]: done

File: ppdet/modeling/bbox_utils.py (hunk 5 above)
Comment (liuhuiCNN) [reply ppdet/modeling/bbox_utils.py]: done

File: ppdet/modeling/heads/s2anet_head.py (hunk 7 above)
Comment (liuhuiCNN) [reply ppdet/modeling/heads/s2anet_head.py]: done

File: dataset/dota_coco/dota_generate_test_result.py (hunk 2 above)
Comment (liuhuiCNN) [reply dataset/dota_coco/dota_generate_test_result.py]: done

File: ppdet/modeling/bbox_utils.py (hunk 4 above)
Comment (liuhuiCNN) [reply ppdet/modeling/bbox_utils.py]: done

File: ppdet/modeling/bbox_utils.py (hunk 6 above)
Comment (liuhuiCNN) [reply ppdet/modeling/bbox_utils.py]: 动转静时上面报错，所以改成下面的了。

**Review by heavengate — APPROVED** [review]
